docs/patterns/_code-snippets.md

```diff
@@ -0,0 +1,35 @@
+---
+collection: patterns
+title: code snippets
+---
+
+## .code-snippet
+
+Code snippets should be used when presenting the user with a small snippet of code that they will likely want to copy and paste.
+
+```html
+<div class="code-snippet">
+    <input class="code-snippet__input" value="sudo apt-get update" readonly="readonly">
+    <button class="code-snippet__copy-button">Copy to clipboard</button>
+</div>
+```
+
+<div class="code-snippet">
+    <input class="code-snippet__input" value="sudo apt-get update" readonly="readonly">
+    <button class="code-snippet__action">Copy to clipboard</button>
+</div>
+
+## .code-snippet--inverse
+
+
+```html
+<div class="code-snippet code-snippet--inverse">
```

Comment:
I understand but we need to have a standard and we have a number of components built in the single class way. For example buttons. 

We should study the extend issue and work out if we are bloating our generated code. In most cases we want the modifiers to maintain a close relationship to the base pattern.